troubleshooting: add cache types entries (#1751)
iterative/dvc.org: content/docs/user-guide/troubleshooting.md

Make sure your bucket
[exists](https://docs.aws.amazon.com/AmazonS3/latest/user-guide/create-bucket.html)
in the correct `region` and/or `endpointurl` (see `dvc remote modify`).

## Unable to detect cache type {#no-dvc-cache}

> Unable to detect supported link types, as

Reviewer: ```suggestion
Unable to detect supported link types, as the
```

[cache directory](/doc/command-reference/config#cache) doesn't exist. It is

> usually auto-created by commands such as `dvc add/fetch/pull/run/import`, but
> you could create it manually to enable this check.

Reviewer: ```suggestion
usually created automatically by DVC commands that need it, but you could create
it manually to enable this check (e.g. with ` mkdir .dvc/cache`).
```

> usually created automatically by DVC commands that need it, but you can create
> it manually (e.g. ` mkdir .dvc/cache`) to enable this check.

Reviewer: ```suggestion
usually created automatically by DVC commands that need it, but you can create
it manually (e.g. `mkdir .dvc/cache`) to enable this check.
```

Make sure your bucket
[exists](https://docs.aws.amazon.com/AmazonS3/latest/user-guide/create-bucket.html)
in the correct `region` and/or `endpointurl` (see `dvc remote modify`).

## Unable to detect cache type {#no-dvc-cache}

> Unable to detect supported link types, as cache directory `.dvc/cache` doesn't exist. It is usually auto-created by commands such as `dvc add/fetch/pull/run/import`, but you could create it manually to enable this check.

Reviewer 2: ```suggestion
Unable to detect supported link types, as [cache directory](/doc/command-reference/config#cache)
doesn't exist. It is usually auto-created by commands such as `dvc add/fetch/pull/run/import`, but you
could create it manually to enable this check.
```

Make sure your bucket
[exists](https://docs.aws.amazon.com/AmazonS3/latest/user-guide/create-bucket.html)
in the correct `region` and/or `endpointurl` (see `dvc remote modify`).

> ## Unable to detect cache type {#unable-to-detect-cache-type}

Reviewer 2: ```suggestion
## Unable to detect cache type {#no-dvc-cache}
```

> Unable to detect supported link types, as cache directory `.dvc/cache` doesn't exist. It is usually auto-created by commands such as `dvc add/fetch/pull/run/import`, but you could create it manually to enable this check.

Reviewer: Thanks! A bit confused. Is the problem that the cache doesn't exist yet, or that the cache type can't be detected? I get that the former causes the latter, but they're also separate situations no? For example, maybe a corrupted DVC cache or bad config file could also cause an inability to detect the type.

Reviewer 2: > For example, maybe a corrupted DVC cache or bad config file could also cause an inability to detect the type.

@jorgeorpinel Not really, we don't use cache files or configs. The only issue here is nonexistent cache dir, that's it.

Reviewer: OK. Still seems kind of indirect but if that's the only possible scenario for this issue, then fine 🙂
